### PR TITLE
Add transformation

### PR DESCRIPTION
**Pull request by IncubatorShokuhou**
#### Describe the changes you've made
Add transformation

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots

 Original           | Updated
 :--------------------: |:--------------------:
 **original screenshot**  | <b>updated screenshot </b> |

### CODE REVIEW

**Comment by IncubatorShokuhou** [comment]
@tvdboom Please review.

File: pycaret/internal/pycaret_experiment/tabular_experiment.py [hunk 1]
```diff
@@ -1106,6 +1118,28 @@ def setup(
 
             self._internal_pipeline.steps.append(("normalize", normalize_estimator))
 
+        # transformation ========================================= >>
+
+        if transformation:
+            if transformation_method == "yeo-johnson":
+                transformation_estimator = PowerTransformer(
+                    method="yeo-johnson", standardize=True, copy=True
```
Comment (tvdboom) [inline pycaret/internal/pycaret_experiment/tabular_experiment.py:1126]: I would suggest to set `standardize=False` and add this transfomation step before normalization. So, if the user wants to normalize after this, he can just use the `normalize` parameter (see #1453)

Comment (tvdboom) [reply pycaret/internal/pycaret_experiment/tabular_experiment.py:1126]: Additionally, we could add box-cox as an additional option

Comment (IncubatorShokuhou) [reply pycaret/internal/pycaret_experiment/tabular_experiment.py:1126]: In fact, yeo-johnson is an extension of box-cox over the real number field (see Johnson, Y. R. A. . (2000). A new family of power transformations to improve normality or symmetry. Biometrika, 87(4), 954-959. ). Therefore, I suggest that we implement the auto-selection of box-cox and yeo-johnson. When the label is all positiveuse, use box-cox to speed up the calculation, and automatically switch to yeo-johnson when the label is not all positive.

Comment (tvdboom) [reply pycaret/internal/pycaret_experiment/tabular_experiment.py:1126]: The transformation is applied on every column in the dataset, so we would need to check if all columns are strictly positive or make two lists of columns (one strictly positive and the other not) and add a columntransformer containing two estimators, one for each feature group. Besides, it still could fail if the user wants to apply the pipeline to new data, without realizing it contains negative values in a column.

Comment (tvdboom) [reply pycaret/internal/pycaret_experiment/tabular_experiment.py:1126]: In the spirit of simplicity, maybe we should leave it as is. Box-cox is almost never used (as far as I know) and this way we don't need to change the API. Only change the normalization part I mentioned in the first comment. What do you think?

Comment (IncubatorShokuhou) [reply pycaret/internal/pycaret_experiment/tabular_experiment.py:1126]: Yes, I totally agree with you. I didn't think about the test dataset before.

Comment (IncubatorShokuhou) [reply pycaret/internal/pycaret_experiment/tabular_experiment.py]: I have revised it according to your first comment.

Comment (IncubatorShokuhou) [reply pycaret/internal/pycaret_experiment/tabular_experiment.py]: BTW, there seems to be an indentation error (`transformation` et.al. should be behind `Preprocessing`). Maybe you can help review it to see if I fixed it correctly.

Comment (tvdboom) [reply pycaret/internal/pycaret_experiment/tabular_experiment.py]: looks good. accepting the PR

**Review by tvdboom — APPROVED** [review]

**Pull request merged**